src/components/ui/dashboard/coaching-session-dialog.tsx

```diff
@@ -0,0 +1,150 @@
+  );
+  const [sessionTime, setSessionTime] = useState<string>(
+    existingSession ? format(new Date(existingSession.date), "HH:mm") : ""
+  );
+  const { isCoach } = useAuthStore((state) => state);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!sessionDate || !sessionTime) return;
+
+    const [hours, minutes] = sessionTime.split(":").map(Number);
+    const dateTime = getDateTimeFromString(sessionDate.toISOString())
+      .set({ hour: hours, minute: minutes })
+      .toFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+    try {
+      if (mode === "create") {
+        const newCoachingSession: CoachingSession = {
+          ...defaultCoachingSession(),
+          coaching_relationship_id: currentCoachingRelationshipId,
+          date: dateTime,
+        };
+        await createCoachingSession(newCoachingSession);
+      } else if (existingSession) {
+        await update(existingSession.id, {
+          ...existingSession,
+          date: dateTime,
+          updated_at: DateTime.now(),
+        });
+      }
+      refresh();
+      onCoachingSessionUpdated();
+      setSessionDate(undefined);
+      setSessionTime("");
+      onOpenChange(false);
+    } catch (error) {
+      console.error(`Failed to ${mode} coaching session:`, error);
+    }
+  };
+
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      {dialogTrigger && (
+        <DialogTrigger asChild>
+          {React.cloneElement(dialogTrigger, {
+            ...(dialogTrigger.props as React.HTMLAttributes<HTMLButtonElement>),
+            className: cn(
+              (dialogTrigger.props as React.HTMLAttributes<HTMLButtonElement>)
+                .className
+            ),
+          })}
+        </DialogTrigger>
+      )}
+
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>
+            {mode === "create"
+              ? "Create New Coaching Session"
+              : "Update Coaching Session"}
+          </DialogTitle>
+        </DialogHeader>
+        <form onSubmit={handleSubmit} className="space-y-4">
+          <div className="space-y-2">
+            <Label htmlFor="session-date">Session Date</Label>
+            <Calendar
+              mode="single"
+              selected={sessionDate}
+              onSelect={(date) => setSessionDate(date)}
+            />
+          </div>
+          <div className="space-y-2">
+            <Label htmlFor="session-time">Session Time</Label>
+            <input
+              type="time"
+              id="session-time"
+              value={sessionTime}
+              onChange={(e) => setSessionTime(e.target.value)}
+              className="w-full border rounded p-2"
+              required
+            />
+          </div>
+          <Button type="submit" disabled={!sessionDate || !sessionTime}>
+            {mode === "create" ? "Create Session" : "Update Session"}
+          </Button>
+        </form>
```

Comment:
We probably want the form itself (and its submit logic) moved into it's own react component and then composed into the dialog component. Separates concerns of the dialog UI vs the form pretty nicely that way.

```
<CoachingSessionDialog ...props >
    <CoachingSessionForm ...props />
</CoachingSessionDialog>
```

Reply:
I think that's a sensible suggestion Zach.